src/WebJobs.Extensions.DurableTask/Correlation/TelemetryActivator.cs

```diff
@@ -48,6 +48,12 @@ public ValueTask DisposeAsync()
             return this.telemetryModule?.DisposeAsync() ?? default;
         }
 
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            this.telemetryModule?.DisposeAsync().AsTask().GetAwaiter().GetResult();
+        }
+
```

Comment:
I'm a bit concerned about this pattern, as I understand that calling `GetAwaiter()` is generally anti-pattern. Do we use disposing pattern (`.GetAwaiter().GetResult`) anywhere else in the codebase?

Reply:
Yes, I see it used in DurableTaskExtension.cs ([example](https://github.com/Azure/azure-functions-durable-extension/blob/18d85bd8c0d41d5936cac46054e3839066ef65a7/src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs#L559))  and DurableTaskListener.cs ([example](https://github.com/Azure/azure-functions-durable-extension/blob/18d85bd8c0d41d5936cac46054e3839066ef65a7/src/WebJobs.Extensions.DurableTask/Listener/DurableTaskListener.cs#L88)).